### PR TITLE
Support gradient clipping in TFPark

### LINKED ISSUE #1281: TFEstimatorSpec cannot support gradient clipping (and other customized train_ops)
Currently `TFEstimatorSpec` can only support construct from `loss` when training.  This might cause trouble when migrating some cases with **customized training ops**, like **gradient clipping** used in [chines_ocr](https://github.com/YCG09/chinese_ocr/blob/master/ctpn/lib/fast_rcnn/train.py#L106-L111) as follows:
```python
tvars = tf.trainable_variables()
grads, norm = tf.clip_by_global_norm(tf.gradients(total_loss, tvars), 10.0)
train_op = opt.apply_gradients(list(zip(grads, tvars))
sess.run(train_op)
```
Maybe we might support passing `train_op` into `TFEstimatorSpec` just like [tf.estimator.EstimatorSpec](https://www.tensorflow.org/api_docs/python/tf/estimator/EstimatorSpec) ? Or we might at least support **gradient clipping** so that migration of `chinese_ocr` could be correct.

### PR DESCRIPTION
**Pull request by yangw1234**
fixed #1281
In the keras case, if user set `clipNorm` and/or `clipValue` in tf.keras.Optimizers, these values will be set automatically.
In the estimator case, user use
```
estimator.set_constant_gradient_clipping(min, max)
estimator.set_gradient_clipping_by_l2_norm(clip_norm)
```

### CODE REVIEW

File: pyzoo/zoo/pipeline/api/net/tf_optimizer.py [hunk 1]
```diff
@@ -200,6 +203,12 @@ def to_floats(vs):
                                               batch_size=batch_size,
                                               optim_method=self.optim_method)
 
+        if self.clip_norm:
+            self.optimizer.set_gradclip_l2norm(self.clip_norm)
+        if self.clip_constant:
+            min_value, max_value = self.clip_constant
```
Comment (hkvision) [inline pyzoo/zoo/pipeline/api/net/tf_optimizer.py]: Add some document or assert to tell users that clip_value should be a tuple?

File: pyzoo/zoo/pipeline/api/net/tf_optimizer.py [hunk 2]
```diff
@@ -227,12 +236,15 @@ def _get_arguments_from_loss(loss, optim_method, session, val_outputs, val_label
 
     @classmethod
     def from_loss(cls, loss, optim_method, session=None, val_outputs=None,
-                  val_labels=None, val_method=None, val_split=0.0, **kwargs):
+                  val_labels=None, val_method=None, val_split=0.0,
+                  clip_norm=None, clip_value=None, **kwargs):
```
Comment (hkvision) [inline pyzoo/zoo/pipeline/api/net/tf_optimizer.py]: is this needed here? as users can always specify them in `kwargs`

File: pyzoo/zoo/pipeline/api/net/tf_optimizer.py (hunk 1 above)
Comment (yangw1234) [reply pyzoo/zoo/pipeline/api/net/tf_optimizer.py]: sure

File: pyzoo/zoo/pipeline/api/net/tf_optimizer.py [hunk 3]
```diff
@@ -58,7 +58,8 @@ class TFOptimizer:
     def __init__(self, loss, optim_method, sess=None, dataset=None, inputs=None,
                  grads=None, variables=None, graph=None,
                  val_outputs=None, val_labels=None, val_method=None, val_split=0.0,
-                 tensors_with_value=None, session_config=None):
+                 tensors_with_value=None, session_config=None,
+                 clip_norm=None, clip_value=None):
```
Comment (jason-dai) [inline pyzoo/zoo/pipeline/api/net/tf_optimizer.py]: Why don't we use `set_constant_gradient_clipping` or `set_gradient_clipping_by_l2_norm` for `TFOptimizer`?

**Comment by hkvision** [comment]
LGTM

File: pyzoo/zoo/pipeline/api/net/tf_optimizer.py (hunk 3 above)
Comment (yangw1234) [reply pyzoo/zoo/pipeline/api/net/tf_optimizer.py]: this is for the keras case, in which the gradient clipping information is hidden in the keras.optimizer

File: pyzoo/zoo/pipeline/api/net/tf_optimizer.py (hunk 2 above)
Comment (yangw1234) [reply pyzoo/zoo/pipeline/api/net/tf_optimizer.py]: when they jump to the source, they will see this option clearly

File: pyzoo/zoo/pipeline/api/net/tf_optimizer.py (hunk 3 above)
Comment (jason-dai) [reply pyzoo/zoo/pipeline/api/net/tf_optimizer.py]: But users of TFOptimizer can also use the gradient clipping support if they want to, yes?

Comment (yangw1234) [reply pyzoo/zoo/pipeline/api/net/tf_optimizer.py]: I'll add those methods. Just override the constructor arguments.

**Comment by yangw1234** [comment]
jenkins pass http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/2329/
